src/index.py

```diff
@@ -269,35 +304,39 @@ def _check_contents_item(
         )
 
     # Check whether item is hidden and a directory
+    item_reference_type = classify_item_reference(
+        reference=item.reference_value, docs_path=docs_path
+    )
     item_path = docs_path / Path(item.reference_value)
-    if item.hidden and item_path.is_dir():
+    if item.hidden and item_reference_type == ItemReferenceType.DIR:
         raise InputError(f"A hidden item is a directory. {item=!r}")
 
-    # Check that the next item is within the directory
-    item_relative_path = Path(item.reference_value)
-    try:
-        item_to_aggregate_path = item_relative_path.relative_to(aggregate_dir)
-    except ValueError as exc:
-        raise InputError(
-            "A nested item is a reference to a path that is not within the directory of its "
-            f"parent. {item=!r}, expected parent path: {aggregate_dir!r}"
-        ) from exc
-
-    # Check that the item is directly within the current directory
-    if len(item_to_aggregate_path.parents) != 1:
-        raise InputError(
-            "A nested item is a reference to a path that is not immediately within the "
-            f"directory of its parent. {item=!r}, expected parent path: {aggregate_dir!r}"
-        )
+    if item_reference_type in {ItemReferenceType.DIR, ItemReferenceType.FILE}:
+        # Check that the next item is within the directory
+        item_relative_path = Path(item.reference_value)
+        try:
+            item_to_aggregate_path = item_relative_path.relative_to(aggregate_dir)
+        except ValueError as exc:
+            raise InputError(
+                "A nested item is a reference to a path that is not within the directory of its "
+                f"parent. {item=!r}, expected parent path: {aggregate_dir!r}"
+            ) from exc
 
-    # Check that if the item is a file, it has the correct extension
-    if item_path.is_file():
-        if item_path.suffix.lower() != DOC_FILE_EXTENSION:
+        # Check that the item is directly within the current directory
+        if len(item_to_aggregate_path.parents) != 1:
             raise InputError(
-                "An item in the contents list is not of the expected file type. "
-                f"{item=!r}, expected extension: {DOC_FILE_EXTENSION}"
+                "A nested item is a reference to a path that is not immediately within the "
+                f"directory of its parent. {item=!r}, expected parent path: {aggregate_dir!r}"
             )
 
+        # Check that if the item is a file, it has the correct extension
+        if item_path.is_file():
```

Comment:
Shouldn't we use `item_reference_type == ItemReferenceType.FILE` instead ?

Reply:
thanks, should be fixed now